src/services/sync/strategies/CSVSyncer.js

```diff
@@ -135,13 +136,13 @@ class CSVSyncer extends Syncer {
 				throw new Error(`Parsing failed. Expected attribute "${param}"`);
 			}
 		});
-		// validate optional params:
-		if (records[0].class !== undefined) {
-			this.importClasses = true;
-		}
 		return records;
 	}
 
+	static needsToImportClasses(records) {
+		return records[0].class !== undefined;
```

Comment:
check array contains at least one element if accessing [0]

Reply:
This code is guarded by https://github.com/schul-cloud/schulcloud-server/pull/674/files/0ebe543f1d0f7afa2928db7f7e9c2482ab0ea78b#diff-53d84f5fa2f6f6d0b028d6a12bc9adaeR119 which checks for the non-array and empty-array cases